### PR TITLE
Add V1 Scenario tests for data transformation

### LINKED ISSUE #2711: Create functional tests for all V1 Data Transformation scenarios
As laid out in #2498 , we need scenarios to cover the Data Transformation functionality we want fully supported in V1.

- Extensible transformation: It should be possible to write simple row-mapping transforms.
Examples: "I can add custom steps to my pipeline such as creating a new column that is the addition of two other columns, or easily add cosine similarity, without having to create my own build of ML.NET.
- I can modify settings in the TextFeaturizer to update the number of word-grams and char-grams used along with things like the normalization.	 	 	 
- I can apply normalization to the columns of my data

### PR DESCRIPTION
**Pull request by rogancarr**
Fixes #2711 
Towards #2498 

* Extensible transformation: It should be possible to write simple row-mapping transforms.
  Examples: "I can add custom steps to my pipeline such as creating a new column that is the addition of two other columns, or easily add cosine similarity, without having to create my own build of ML.NET.
* I can modify settings in the TextFeaturizer to update the number of word-grams and char-grams used along with things like the normalization.
* I can apply normalization to the columns of my data

### CODE REVIEW

File: test/Microsoft.ML.Functional.Tests/Datasets/FeatureColumn.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Microsoft.ML.Functional.Tests.Datasets
+{
+    /// <summary>
+    /// A class to hold a feature column.
+    /// </summary>
+    internal sealed class FeatureColumn
+    {
+        public float[] Features { get; set; }
+    }
```
Comment (sfilipi) [inline test/Microsoft.ML.Functional.Tests/Datasets/FeatureColumn.cs]: maybe move it to the other file, the Iris.

File: test/Microsoft.ML.Functional.Tests/DataTransformation.cs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                Assert.Equal(randomNumber1, row.Float1);
+                Assert.Equal(randomNumber2, row.Float2);
+            }
+        }
+
+        /// <summary>
+        /// Extensibility: Add a new column that is a function of other columns.
+        /// </summary>
+        [Fact]
+        void ExtensibilityAddAColumnAsAFunctionOfMultipleColumns()
```
Comment (sfilipi) [inline test/Microsoft.ML.Functional.Tests/DataTransformation.cs]: >ExtensibilityAddAColumnAsAFunctionOfMultipleColumns [](start = 13, length = 51)

to me this test seems identical to the ExtensibilityAddingOneColumn. The function is different, but does that matter to the CustomMapping?

File: test/Microsoft.ML.Functional.Tests/DataTransformation.cs [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+            var scoredData = model.Transform(data);
+            var metrics = mlContext.BinaryClassification.Evaluate(scoredData);
+
+            // Check that the metrics returned are valid.
+            Common.AssertMetrics(metrics);
+        }
+
+        /// <summary>
+        /// Extensibility: Featurize text using custom word-grams, char-grams, and normalization.
+        /// </summary>
```
Comment (artidoro) [inline test/Microsoft.ML.Functional.Tests/DataTransformation.cs]: Could you change summary tag to match the test?

File: test/Microsoft.ML.Functional.Tests/DataTransformation.cs (hunk 2 above)
Comment (artidoro) [reply test/Microsoft.ML.Functional.Tests/DataTransformation.cs]: I agree, maybe I would only keep one of the two functions.

---
In reply to: [261830627](https://github.com/dotnet/machinelearning/pull/2803#discussion_r261830627) [](ancestors = 261830627)

File: test/Microsoft.ML.Functional.Tests/Datasets/FeatureColumn.cs (hunk 1 above)
Comment (rogancarr) [reply test/Microsoft.ML.Functional.Tests/Datasets/FeatureColumn.cs]: This is designed to be generic for tests against any features columns, not just the iris, so I'll keep it in its own file for now, similar to `FeatureContributionOutput'.

---
In reply to: [261830507](https://github.com/dotnet/machinelearning/pull/2803#discussion_r261830507) [](ancestors = 261830507)

**Pull request merged**